homeassistant/components/ecodevices/sensor.py

```diff
@@ -0,0 +1,263 @@
+                config.get(CONF_T2_UNIT_OF_MEASUREMENT),
+                "power",
+                "mdi:flash",
+            )
+        )
+    if config.get(CONF_C1_ENABLED):
+        _LOGGER.debug("Add the meter 1 entity")
+        entities.append(
+            C1EdDevice(
+                controller,
+                coordinator,
+                "c1",
+                config.get(CONF_C1_NAME),
+                config.get(CONF_C1_UNIT_OF_MEASUREMENT),
+                config.get(CONF_C1_DEVICE_CLASS),
+            )
+        )
+    if config.get(CONF_C2_ENABLED):
+        _LOGGER.debug("Add the meter 2 entity")
+        entities.append(
+            C2EdDevice(
+                controller,
+                coordinator,
+                "c2",
+                config.get(CONF_C2_NAME),
+                config.get(CONF_C2_UNIT_OF_MEASUREMENT),
+                config.get(CONF_C2_DEVICE_CLASS),
+            )
+        )
+
+    if entities:
+        async_add_entities(entities, True)
+
+
+class EdDevice(CoordinatorEntity):
+    """Representation of a generic Eco-Devices sensor."""
+
+    def __init__(
+        self,
+        controller,
+        coordinator,
+        input_name,
+        name,
+        unit,
+        device_class,
+        icon=None,
+    ):
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self.controller = controller
+        self._input_name = input_name
+        self._name = name
+        self._unit = unit
+        self._device_class = device_class
+        self._icon = icon
+        self._state = None
+
+    @property
+    def device_info(self):
+        """Return device information identifier."""
+        return {
+            "identifiers": {(DOMAIN, self.controller.host)},
+            "via_device": (DOMAIN, self.controller.host),
+        }
+
+    @property
+    def unique_id(self):
+        """Return an unique id."""
+        return "_".join(
+            [
+                DOMAIN,
+                self.controller.host,
+                "sensor",
+                self._input_name,
+            ]
+        )
+
+    @property
+    def device_class(self):
+        """Return the device_class."""
+        return self._device_class
+
+    @property
+    def name(self):
+        """Return the name."""
+        return self._name
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit_of_measurement if specified."""
+        return self._unit
+
+    @property
+    def icon(self):
+        """Return the icon if specified."""
+        return self._icon
+
+
+class T1EdDevice(EdDevice):
+    """Initialize the T1 sensor."""
+
+    @property
+    def state(self):
+        """Return the state."""
+        return self.coordinator.data["T1_PAPP"]
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes."""
+        if self.coordinator.data:
+            return {
+                "type_heures": self.coordinator.data.get("T1_PTEC"),
```

Comment:
For what are these values required? Shouldn't they be separate sensor entities if this is important.

Reply:
They are plenty values available, and each person are not interested by all of them, so I choosed to only create one sensor on the main value and put all others in attributes, to avoid getting a lot of sensor, but maybe it's not the good solution ?